yjit/src/options.rs

```diff
@@ -128,6 +137,14 @@ pub enum TraceExits {
     Counter(Counter),
 }
 
+#[derive(Clone, Copy, PartialEq, Eq, Debug)]
+pub enum CompilationLogOutput {
+    // Dump to stdout when the process exits
+    Stdout,
+    // Dump to the log file as compilation events occur.
```

Comment:
I expected the compilation log to be printed as it compiles. When you use the `Stdout` mode and compilation logs are printed in between yjit-bench iterations, you can know which methods are long-tail problems. But if it's only printed at the end, it's harder to understand when things are compiled. I would also like `File` to print logs as it compiles so that I can tail the file.

Basically, I don't get why you need to buffer outputs and print all of them at the end. The feature seems closer to `--yjit-dump-disasm` (which doesn't print stuff at exit) than `--yjit-stats` (which prints stats at exit).

Reply:
It now prints to `$stderr` as compilation occurs. That obviated the need for this enum entirely, so I've also removed that.

Reply:
Actually, we still have the enum. It turns out that `std::os:fd::AsRawFd` wasn't stabilized until Rust 1.66.0. I've changed my local Rust to 1.58.0 to avoid surprises like that going forward.